gr-zeromq/lib/pull_msg_source_impl.cc

```diff
@@ -91,8 +91,12 @@ void pull_msg_source_impl::readloop()
 
             std::string buf(static_cast<char*>(msg.data()), msg.size());
             std::stringbuf sb(buf);
-            pmt::pmt_t m = pmt::deserialize(sb);
-            message_port_pub(d_port, m);
+            try {
+                pmt::pmt_t m = pmt::deserialize(sb);
```

Comment:
I like this change, but should the try be isolated to the pmt::deserialize - i.e. will message_port_pub throw a pmt exception?

Reply:
The `message_port_pub` does not throw `pmt::exception` but I think it's better to publish/post only if the PMT message was deserialized correctly and that's why I put `message_port_pub` inside the try block.

I can rewrite it as
```cpp
            pmt::pmt_t m;
            try {
                m = pmt::deserialize(sb);
            } catch (pmt::exception& e) {
                GR_LOG_ERROR(d_logger, std::string("Invalid PMT message: ") + e.what());
                continue;
            }
            message_port_pub(d_port, m);
```
but I found this less readable. Let me know if you like more this variant or if you have another idea and I'm going to change it.

Reply:
True - I agree with you there.  The original change is better